sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicy.java

```diff
@@ -98,6 +101,37 @@ public OpenTelemetryHttpPolicy() {
                 .contextWrite(reactor.util.context.Context.of(REACTOR_PARENT_TRACE_CONTEXT_KEY, startSpan(context)));
     }
 
+    @Override
+    public HttpResponse processSynchronously(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        if ((boolean) context.getData(DISABLE_TRACING_KEY).orElse(false)) {
+            return next.processSynchronously();
+        }
+
+        Context telemetryContext = startSpan(context);
+        Object agentContext = OpenTelemetrySpanSuppressionHelper.registerClientSpan(telemetryContext);
+        AutoCloseable closeable = OpenTelemetrySpanSuppressionHelper.makeCurrent(agentContext, telemetryContext);
+        HttpResponse responseToBeRecorded = null;
+        RuntimeException exception = null;
+        try {
+            HttpResponse response = next.processSynchronously();
+            responseToBeRecorded = response;
+            return response;
+        } catch (RuntimeException e) {
+            exception = e;
+            if (e instanceof HttpResponseException) {
+                responseToBeRecorded = ((HttpResponseException) e).getResponse();
+            }
+            throw LOGGER.logExceptionAsError(e); // TODO (kasobol-msft) this probably shouldn't log here.
```

Comment:
Agreed, this shouldn't go here and I also believe an `HttpResponseException` won't be thrown in this scenario as those are created in `RestProxy` which is called at the end of the pipeline execution

Reply:
I'll keep this one as todo in case this code moves more around.

Reply:
how does the `HttpResponseException` work in the async case then ?